### ISSUE TITLE
SENSEMBED: Learning Sense Embeddings for Word and Relational Similarity

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/deep-learning-readinglist#91]
http://wwwusers.di.uniroma1.it/~navigli/pubs/ACL_2015_Iacobaccietal.pdf

**Comment by sweaterr** [comment]
많이 축하 된 단어 임베딩은 단어가 나타날 것으로 예상되는 컨텍스트를 모델링하기 위해 신경망을 활용하는 코퍼스 기반의 분산 시맨틱 모델의 새로운 지점을 나타냅니다. 높은 통찰력과 구문 및 의미 정보를 모두 포착 할 수있는 능력 덕분에 워드 임베디드는 Word Sense Disambiguation (Chen et al., 2014), 기계 번역 (Mikolov et al. , 2013b), 관계 적 유사성 (Mikolov et al., 2013c), 의미 론적 연관성 (Baroni et al., 2014) 및 지식 표현 (Bordes et al., 2013) 그러나 단어 삽입은 이전의 집단 기반 분산 모델에서 두 가지 중요한 제한을 상속받습니다. (1)

**Comment by sweaterr** [comment]
몇 가지 연구 작업이 이러한 문제를 해결하기 위해 노력했습니다. 예를 들어, Reisinger and Mooney (2010)의 원래 감각 차별 접근법에 기초하여 연구를 진행 한 Huang et al. (2012)는 K-means 클러스터링을 적용하여 단어 삽입을 여러 프로토 타입으로 분해합니다. 각 프로토 타입은 대상 단어의 고유 한 의미를 나타냅니다. 그러나 얻은 감각 표현은 어떤 감각 인벤토리와도 연결되지 않으므로 결과적으로 수동으로 또는 감성 주석 데이터를 사용하여 매핑해야합니다. 연구의 또 다른 라인은 단어 임베딩에서 기존의 의미 론적 자원을 이용할 가능성을 조사합니다. 좋은 예는 관계 제약 모델 (Yu and Dredze, 2014)입니다. 단어 삽입을 계산할 때,이 모델은 원본 코퍼스의 원본 동시 발생 단서를 의역 구문 데이터베이스의 자동 추출 된 데이터 집합 인 Paraphrase Database1 (Ganitkevitch et al., 2013, PPDB)에서 파생 된 관계 정보로 바꿉니다.

**Comment by sweaterr** [comment]
However, none of these techniques have simultaneously solved both above-mentioned issues, i.e., inability to model polysemy and reliance on text corpora as the only source of knowledge. We propose a novel approach, called SENSEMBED, which addresses both drawbacks by exploiting semantic knowledge for modeling arbitrary word senses in a large sense inventory. We evaluate our representation on multiple datasets in two standard tasks: word-level semantic similarity and relational similarity. Experimental results show that moving from words to senses, while making use of lexical-semantic knowledge bases, makes embeddings significantly more powerful, resulting in consistent performance improvement across tasks.

그러나 이러한 기법들 중 어느 것도 앞서 언급 한 두 가지 문제, 즉 polysemy을 모델화 할 수없고  지식의 유일한 원천으로 텍스트 코퍼스에 의존하는 문제를 동시에 해결하지 못했습니다. 우리는 SENSEMBED라는 새로운 접근 방법을 제안합니다.이 접근 방법은 큰 의미의 인벤토리에서 임의의 단어 감각을 모델링하기 위해 의미론적 지식을 활용함으로써 두 가지 단점을 해결합니다. 우리는 두 가지 표준 작업, 즉 단어 수준 의미 론적 유사성과 관계형 유사성에서 여러 데이터 집합에 대한 표현을 평가합니다. 실험 결과에 따르면 어휘 - 의미 론적 지식 기반을 사용하면서 단어에서 감각으로 이동하면 임베딩이 훨씬 더 강력 해져 작업 전반에 걸쳐 일관된 성능 향상을 보입니다.

**Comment by sweaterr** [comment]
Word embeddings are vector space models (VSM) that represent words as real-valued vectors in a low-dimensional (relative to the size of the vocabulary) semantic space, usually referred to as the continuous space language model. The conventional way to obtain such representations is to compute a term-document occurrence matrix on large corpora and then reduce the dimensionality of the matrix using techniques such as singular value decomposition (Deerwester et al., 1990; Bullinaria and Levy, 2012, SVD). Recent predictive techniques (Bengio et al., 2003; Collobert and Weston, 2008; Mnih and Hinton, 2007; Turian et al., 2010; Mikolov et al., 2013a) replace the conventional two-phase approach with a single supervised process, usually based on neural networks. In contrast to word embeddings, which obtain a single model for potentially ambiguous words, sense embeddings are continuous representations of individual word senses. In order to be able to apply word embeddings techniques to obtain representations for individual word senses, large sense-annotated corpora have to be available. However, manual sense annotation is a difficult and time-consuming process, i.e., the so-called knowledge acquisition bottleneck. In fact, the largest existing manually sense annotated dataset is the SemCor corpus (Miller et al., 1993), whose creation dates back to more than two decades ago. In order to alleviate this issue, we leveraged a state-of-the-art Word Sense Disambiguation (WSD) algorithm to automatically generate large amounts of sense-annotated corpora.

중의성을 고려한 단어 임베딩 생성은 어렵다. 왜냐하면, annotate된 데이터를 얻기 어렵기 때문이다.

**Comment by sweaterr** [comment]
섹션 2의 나머지 부분에서, 먼저 2.1 절에서 SENSEMBED에 사용 된 센스 인벤토리를 설명합니다. 2.2 절은 코퍼스와이 코퍼스에 주석을 달기 위해 사용 된 모호성 제거 절차를 소개한다. 마지막으로 섹션 2.3에서 우리는 감각 삽입의 훈련을 위해 자동 감각 태그가 붙은 데이터 세트를 활용하는 방법을 논의합니다.

**Comment by sweaterr** [comment]
2.1 Underlying sense inventory
We selected BabelNet2 (Navigli and Ponzetto, 2012) as our underlying sense inventory. The resource is a merger of WordNet with multiple other lexical resources, the most prominent of which is Wikipedia. As a result, the manually-curated information in WordNet is augmented with the complementary knowledge from collaborativelyconstructed resources, providing a high coverage of domain-specific terms and named entities and a rich set of relations. The usage of BabelNet as our underlying sense inventory provides us with the advantage of having our sense embeddings readily applicable to multiple sense inventories.
한국어
BabelNet2 (Navigli and Ponzetto, 2012)를 기본 감각 인벤토리로 선택했습니다. 이 자원은 WordNet과 여러 다른 어휘 자원 (Wickipedia) 중 가장 두드러진 것이 WordNet의 합병입니다. 결과적으로 WordNet의 수동으로 큐 레이팅 된 정보는 협업 적으로 구성된 리소스의 보완 지식으로 보강되어 도메인 관련 용어 및 명명 된 엔티티와 풍부한 관계 세트를 제공합니다. BabelNet을 기본 감각 인벤토리로 사용하면 감각 매립을 여러 감각 인벤토리에 쉽게 적용 할 수 있다는 장점이 있습니다.

**Comment by sweaterr** [comment]
2.2 Generating a sense-annotated corpus
As our corpus we used the September-2014 dump of the English Wikipedia.3 This corpus comprises texts from various domains and topics and provides a suitable word coverage. The unprocessed text of the corpus includes approximately three billion tokens and more than three million unique words. We only consider tokens with at least five occurrences.
한국어
우리 코퍼스는 영어권 위키피디아의 9 월 2014 일 덤프를 사용했습니다 .3이 코퍼스는 다양한 도메인과 주제의 텍스트로 구성되어 있으며 적절한 단어 범위를 제공합니다. 코퍼스의 처리되지 않은 텍스트에는 약 30 억 개의 토큰과 3 백만 개 이상의 고유 단어가 포함됩니다. 우리는 적어도 다섯 번 발생하는 토큰만을 고려합니다.

**Comment by sweaterr** [comment]
As our WSD system, we opted for Babelfy4 (Moro et al., 2014), a state-of-the-art WSD and Entity Linking algorithm based on BabelNet’s semantic network. Babelfy first models each concept in the network through its corresponding “semantic signature” by leveraging a graph random walk algorithm. Given an input text, the algorithm uses the generated semantic signatures to construct a subgraph of the semantic network representing the input text. Babelfy then searches this subgraph for the intended sense of each content word using an iterative process and a dense subgraph heuristic. Thanks to its use of BabelNet, Babelfy inherently features multilinguality; hence, our representation approach is equally applicable to languages other than English. In order to guarantee high accuracy and to avoid bias towards more frequent senses, we do not consider those judgements made by Babelfy while backing off to the most frequent sense, a case that happens when a certain confidence threshold is not met by the algorithm. The disambiguated items with high confidence correspond to more than 50% of all the content words. As a result of the disambiguation step, we obtain sense-annotated data comprising around one billion tagged words with at least five occurrences and 2.5 million unique word senses.
한국어
WSD 시스템으로 BabelNet의 의미 네트워크를 기반으로 한 최첨단 WSD 및 엔티티 링크 알고리즘 인 Babelfy4 (Moro et al., 2014)를 선택했습니다. Babelfy는 먼저 그래프 랜덤 보행 알고리즘을 활용하여 해당 의미 론적 서명을 통해 네트워크의 각 개념을 모델링합니다. 입력 텍스트가 주어지면 알고리즘은 생성 된 의미 론적 서명을 사용하여 입력 텍스트를 나타내는 시맨틱 네트워크의 서브 그래프를 구성합니다. 그런 다음 Babelfy는이 하위 그래프에서 반복적 인 프로세스와 밀도가 높은 서브 그래프 경험적 방법을 사용하여 각 콘텐츠 단어의 의도 된 의미를 검색합니다. BabelfNet의 사용 덕분에 Babelfy는 근본적으로 다 언어 성을 특징으로합니다. 그러므로 우리의 대행 방식은 영어 이외의 언어에도 똑같이 적용됩니다. 높은 정확성을 보장하고보다 빈번한 감각에 편견을 피하기 위해, Babelfy가 가장 빈번한 감각으로 되돌아가는 동안 알고리즘에 의해 특정 신뢰 임계 값이 충족되지 않을 때 발생하는 판단을 고려하지 않습니다. 높은 신뢰도를 가진 모호하지 않은 항목은 모든 콘텐츠 단어의 50 % 이상에 해당합니다. disambiguation 단계의 결과로, 우리는 적어도 5 개의 사건과 250 만 개의 고유 한 단어 감각을 가진 약 10 억 개의 태그가있는 단어로 구성된 감각 - 주석 데이터를 얻습니다.

**Comment by sweaterr** [comment]
2.3 Learning sense embeddings

The disambiguated text is processed with the Word2vec (Mikolov et al., 2013a) toolkit5 . We applied Word2vec to produce continuous representations of word senses based on the distributional information obtained from the annotated corpus. For each target word sense, a representation is computed by maximizing the log likelihood of the word sense with respect to its context. We opted for the Continuous Bag of Words (CBOW) architecture, the objective of which is to predict a single word (word sense in our case) given its context. The context is defined by a window, typically with the size of five words on each side with the paragraph ending barrier. We used hierarchical softmax as our training algorithm. The dimensionality of the vectors were set to 400 and the subsampling of frequent words to 10−3 .
한국어
명확성이 보장 된 텍스트는 Word2vec (Mikolov 외, 2013a) 툴킷 5으로 처리됩니다. 우리는 Word2vec를 사용하여 주석 처리 된 코퍼스에서 얻은 분포 정보를 기반으로 단어 감각을 연속적으로 표현했습니다. 각각의 목표 단어 감각에 대해 표현은 문맥에 대한 단어 감각의 로그 우도를 최대화하여 계산됩니다. 우리는 Continuous Bag of Words (CBOW) 아키텍처를 선택했습니다.이 아키텍처의 목표는 컨텍스트를 고려하여 단일 단어 (우리의 경우 단어 의미)를 예측하는 것입니다. 컨텍스트는 일반적으로 문단 장벽이있는 각면에 5 개 단어의 크기를 갖는 창으로 정의됩니다. 훈련 알고리즘으로 계층 적 softmax를 사용했습니다. 벡터의 차원은 400으로 설정되었고 잦은 단어는 10-3으로 서브 샘플링되었습니다.

**Comment by sweaterr** [comment]
As a result of the learning process, we obtain vector-based semantic representations for each of the word senses in the automatically-annotated corpus. We show in Table 1 some of the closest senses to six sample word senses: the geographical and financial senses of river, the performance and phone number senses of number, and the gang and car senses of hood. 6 As can be seen, sense embeddings can capture effectively the clear distinctions between different senses of a word. Additionally, the closest senses are not necessarily constrained to the same part of speech. For instance, the river sense of bank has the adverbs upstream and downstream and the “move along, of liquid” sense of the verb run among its closest senses.
한국어
학습 과정의 결과로, 우리는 각각의 벡터 기반 의미 론적 표현 자동으로 주석이 달린 신체. 표 1에는 가장 가까운 여섯 가지 샘플 단어 감각 : 지리적 그리고 금융 감각, 강 전화 번호 감각과 갱단 후드의 자동차 감각. 6 보시다시피, embeddings 감지 명확한 구별을 효과적으로 포착 할 수있다. 서로 다른 감각 사이. 또한, 가장 가까운 감각은 반드시 제약을받지 않는다. 같은 부분의 연설로 예를 들어, 은행의 강 감각에는 상류 부사가있다. 하류 및 "액체의 이동" 가장 가까운 감각들 사이에서 움직이는 동사의 감각.

**Comment by sweaterr** [comment]
바벨파이로 중의성 해소 후 워드임베딩 돌린듯
![image](https://user-images.githubusercontent.com/1518919/26941035-ef72455c-4cb7-11e7-972a-386ee2e0d926.png)
